substrate/primitives/runtime/src/generic/unchecked_extrinsic.rs

```diff
@@ -40,8 +41,59 @@ use sp_std::{fmt, prelude::*};
 /// the decoding fails.
 const EXTRINSIC_FORMAT_VERSION: u8 = 4;
 
-/// The `SingaturePayload` of `UncheckedExtrinsic`.
-type UncheckedSignaturePayload<Address, Signature, Extra> = (Address, Signature, Extra);
+/// The `SignaturePayload` of `UncheckedExtrinsic`.
+type UncheckedSignaturePayload<Address, Signature, Extension> = (Address, Signature, Extension);
+
+impl<Address: TypeInfo, Signature: TypeInfo, Extension: TypeInfo> SignaturePayload
+	for UncheckedSignaturePayload<Address, Signature, Extension>
+{
+	type SignatureAddress = Address;
+	type Signature = Signature;
+	type SignatureExtra = Extension;
+}
+
+/// TODO: docs
+#[derive(Eq, PartialEq, Clone, Encode, Decode)]
+pub enum Preamble<Address, Signature, Extension> {
```

Comment:
Same as this, about decoding. Is this meant to be backwards compatible in terms of decode? idk yet, but even if the rest of this file will answer that, I suggest clarifying it here furhter.

Reply:
I think it is good to keep the tests around.